### ISSUE TITLE
Caching problem with renamed and duplicate templates

### ISSUE THREAD

**Issue by sebakerckhof** [Urigo/angular-meteor#888]
What I've done: renamed template `institutes.ng.html` to `choose.ng.html` and created a new `institutes.ng.html`

Angular complains the template can't be found and I find this in the source code:
```js
(function(){

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                     //
// packages/barco_admin/client/institutes/view/template.choose.ng.js                                                   //
//                                                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
                                                                                                                       //
angular.module('angular-meteor').run(['$templateCache', function($templateCache) { 
$templateCache.put('/packages/barco:admin/client/institutes/view/institutes.ng.html',"...html...");
}]);
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

}).call(this);
```

As you can see, this template is still put in the angular template cache with the old name. (template.choose.ng.js  <-> /packages/barco:admin/client/institutes/view/institutes.ng.html )

Might be related to: https://github.com/Urigo/angular-meteor/issues/886

As soon as I make some changes to `choose.ng.html`, this solved the problem. So I guess it's an issue with the caching compiler.

**Comment by MilosStanic** [comment]
Definitely caching compiler issue. I've seen terribly weird behaviour on changing files extensions. I almost gave up. Is there a quick fix to force emptying cache? `meteor clean` or something?

**Comment by sebakerckhof** [comment]
Okay, I think I have a solution. 
What appears to happen is that the caching compiler in angular-meteor extends from `CachingHtmlCompiler` which has the following code:
```js
getCacheKey(inputFile) {
    // Note: the path is only used for errors, so it doesn't have to be part
    // of the cache key.
    return inputFile.getSourceHash();
  }
```

If you rename a file, the cache key doesn't change. A solution would be to use both the filename and the source hash as cache key.

I can make a PR but I don't know if this should be done in the `NgCachingHtmlCompiler` here or in `CachingHtmlCompiler` from Meteor, since this could also be a problem for other compilers that extend from CachingHtmlCompiler...

**Comment by mxab** [comment]
you can try to ```rm -rf .meteor/local/plugin-cache```

**Comment by MilosStanic** [comment]
We will have to wait for @Urigo to comment. But I guess this is a good find.

**Comment by sebakerckhof** [comment]
I figured it made sense to fix this in meteor core, since other compilers might also need this.

**Comment by sebakerckhof** [comment]
I created the PR here, since changing this in meteor core would trigger cache invalidations that are not always necessary for people not using angular-meteor.

**Comment by EvilJohnny** [comment]
I was hoping that was some kind of cache problem, but I think is not the case, at least for #886 

I just did what @mxab suggested, and I have the exactly same results. I also created a new fresh Meteor 1.2 proyect, copied over de code, installed needed packages, etc... and once again, the same result :cry:

**Comment by westor21** [comment]
Currently I have the problem, if I just change one character in my template, I also get the message: "Tryed to load angular more then ones". I noticed in the network panel a full index page load under the name of the template instead of just the template.

**Comment by Urigo** [comment]
Released in 1.2.3
